pkg/providers/tinkerbell/stack/stack.go

```diff
@@ -428,14 +440,18 @@ func (s *Installer) Upgrade(ctx context.Context, bundle releasev1alpha1.Tinkerbe
 			return err
 		}
 	}
-
+	opts := map[string]string{}
+	if s.proxyConfig != nil {
+		opts["NO_PROXY"] = s.generateNoProxyString(s.proxyConfig.NoProxy)
```

Comment:
nit: Let's just call strings.Join directly instead of a separate func
```suggestion
		opts["NO_PROXY"] = strings.Join(s.proxyConfig.NoProxy, ",")
```